apps/examples/st_things/msghandling/sample_msghandling_user.c

```diff
@@ -52,6 +62,29 @@ static int64_t g_color_temp_range[MAX_STR_LEN]      = {0, 100};
 
 static char g_machine_state[MAX_STR_LEN]            = "RUNNING";
 
+static size_t g_mode_cnt                            = 0;
+static char **g_modes                               = NULL;
+static char **g_supported_modes                     = NULL;
+
+void init_modes(void)
+{
+	int idx;
+
+	g_mode_cnt = 1;
+	g_modes = (char **)malloc(sizeof(char*) * 1);
+	g_modes[0] = malloc(sizeof(char) * MAX_STR_LEN);
+	strncpy(g_modes[0], g_machine_state, MAX_STR_LEN);
+
+	g_supported_modes = (char **)malloc(sizeof(char*) * MAX_SUPPORTED_MODES);
```

Comment:
I can't find freeing "g_supported_modes" and "element".

Reply:
This is just a example to show how to handle request message.
g_supported_modes is living during the app running. 
If g_supported_modes is freed, the app will get a strange action.